dlup/annotations.py

```diff
@@ -156,118 +327,158 @@ def __str__(self) -> str:
         except KeyError as e:
-            raise AttributeError(str(e)) from None
+            raise AttributeError(f"{name} attribute not found") from e
 
     def __str__(self) -> str:
         return f"{self.annotation_class}, {self.wkt}"
 
 
-def rescale_geometry(geometry: Union[Point, Polygon], scaling: float | None = None) -> Union[Point, Polygon]:
-    if scaling is None:
-        return geometry
-    if scaling == 1.0:
-        return geometry
-
-    scaled_geometry = shapely.affinity.scale(geometry, scaling, scaling)
-    if isinstance(geometry, Polygon):
-        return Polygon(scaled_geometry, a_cls=geometry.annotation_class)
-
-    if isinstance(geometry, Point):
-        return Point(scaled_geometry, a_cls=geometry.annotation_class)
-
-    raise AnnotationError(f"geometry type {type(geometry)} is not a valid dlup type.")
-
-
 class CoordinatesDict(TypedDict):
     type: str
     coordinates: list[list[list[float]]]
 
 
-def shape(coordinates: CoordinatesDict, label: str, multiplier: float = 1.0) -> list[Polygon | Point]:
+def shape(
+    coordinates: CoordinatesDict,
+    label: str,
+    color: Optional[tuple[int, int, int]] = None,
+    z_index: Optional[int] = None,
+) -> list[Polygon | Point]:
     geom_type = coordinates.get("type", None)
     if geom_type is None:
         raise ValueError("No type found in coordinates.")
     geom_type = geom_type.lower()
 
+    if geom_type in ["point", "multipoint"] and z_index is not None:
+        raise AnnotationError("z_index is not supported for point annotations.")
+
     if geom_type == "point":
-        annotation_class = AnnotationClass(label=label, a_cls=AnnotationType.POINT)
+        annotation_class = AnnotationClass(
+            label=label, annotation_type=AnnotationType.POINT, color=color, z_index=None
+        )
         return [
             Point(
-                np.asarray(coordinates["coordinates"]) * multiplier,
+                np.asarray(coordinates["coordinates"]),
                 a_cls=annotation_class,
             )
         ]
-
     if geom_type == "multipoint":
-        annotation_class = AnnotationClass(label=label, a_cls=AnnotationType.POINT)
-        return [Point(np.asarray(c) * multiplier, a_cls=annotation_class) for c in coordinates["coordinates"]]
+        annotation_class = AnnotationClass(
+            label=label, annotation_type=AnnotationType.POINT, color=color, z_index=None
+        )
+        return [Point(np.asarray(c), a_cls=annotation_class) for c in coordinates["coordinates"]]
 
     if geom_type == "polygon":
-        annotation_class = AnnotationClass(label=label, a_cls=AnnotationType.POLYGON)
-        return [
-            Polygon(
-                np.asarray(coordinates["coordinates"][0]) * multiplier,
-                a_cls=annotation_class,
-            )
-        ]
-
+        _coordinates = coordinates["coordinates"]
+        annotation_type = AnnotationType.BOX if _is_rectangle(_coordinates) else AnnotationType.POLYGON
```

Comment:
This is how QuPath parses it, and it's also needed to be able to go from V7 Darwin <-> GeoJSON. It would mean we would break that behavior. I would be in favor of breaking it here and implement our own annotation format that can support this.

It doesn't lead to undefined behavior. With the new changes in `convert_annotations` it still behaves well as the BOX is also a `dlup.annotations.Polygon`. It might be good to have it as a box everywhere, it could be useful for some checks (but which?).

What do you think?

Reply:
This is no longer an issue with the new changes in `convert_annotations`. If boxes are removed from the `convert_annotations`, boxes will be correctly incorporated in the mask. 

I do think we should be thoughtful with how we handle boxes. One use case would be object detection with (non-rotated) bounding boxes as targets. A user can define a custom `convert_annotations` function, but it might be useful to already provide boxes in the standard `"annotation_data"` dictionary. What do you think would be a good solution for this use case? Perhaps we can introduce a more restrictive annotation type `AnnotationType.BOUNDINGBOX`, which only allows non-rotated boxes.

Reply:
Once we have data with clear boxes, I think we can do something with this to see how to handle those. I would for instance think about an object BoundingBox that returns a tuple[int, int, int, int] rather than directly having these as Polygons.

I just don't have annotations like that yet. Since things we consider are usually rotationally invariant we often use points